Fix GitHub Pages Deployment Workflow

- currently the deploy.yml workflow is erroring at the deploy step because of the Git config is not set up in the pipeline
- to fix I'll have to add the Git config step before the deploy step to include my name and email

```shell
> kk-website@0.1.0 deploy
> gh-pages -d build

ProcessError: Author identity unknown

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: empty ident name (for <runner@fv-az1487-47.vpy2y30pan5uhaa0ktko4c43jh.ex.internal.cloudapp.net>) not allowed

    at ChildProcess.<anonymous> (/home/runner/work/kk-website/kk-website/node_modules/gh-pages/lib/git.js:42:16)
    at ChildProcess.emit (node:events:517:28)
    at maybeClose (node:internal/child_process:1098:16)
    at ChildProcess._handle.onexit (node:internal/child_process:303:5)
Error: Process completed with exit code 1.
```

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -28,6 +28,13 @@ jobs:
 
       # Deploy to GitHub Pages
       - name: Deploy to GitHub Pages
-        run: npm run deploy
+        run: |
+          git config --global user.name $user_name
+          git config --global user.email $user_email
+          git remote set-url origin https://${github_token}@github.com/${repository}
+          npm run deploy
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          user_name: 'github-actions[bot]'
+          user_email: 'github-actions[bot]@users.noreply.github.com'
+          github_token: ${{ secrets.ACTIONS_DEPLOY_ACCESS_TOKEN }}
+          repository: ${{ github.repository }}

